src/dmtcpplugin.cpp

```diff
@@ -209,19 +207,17 @@ dmtcp_set_ckpt_file(const char *filename)
 EXTERNC const char *
 dmtcp_get_ckpt_filename(void)
 {
-  static string filename;
-
-  filename = ProcessInfo::instance().getCkptFilename();
-  return filename.c_str();
+  static string *filename =
+    new string(ProcessInfo::instance().getCkptFilename());
+  return filename->c_str();
 }
 
 EXTERNC const char *
 dmtcp_get_ckpt_files_subdir(void)
 {
-  static string tmpdir;
-
-  tmpdir = ProcessInfo::instance().getCkptFilesSubDir();
-  return tmpdir.c_str();
+  static string *tmpdir =
+    new string(ProcessInfo::instance().getCkptFilesSubDir());
```

Comment:
Doesn't this create a memory leak by allocating in the heap, instead of tmpdir on the stack?